resources/index.php

```diff
@@ -152,6 +152,15 @@
 $headers[] = 'Authorization: Bearer {adminApiToken}';
 $headers[] = 'Content-Type: application/x-www-form-urlencoded';
 curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+curl_setopt($ch, CURLOPT_HEADERFUNCTION,
```

Comment:
Hey Simon, altering the Content-Type header overwrites the default header which is:

> text/html; charset=UTF-8

So if you overwrite the content-type header, PowerShell does not know how to interpret the bytes it receives from the HTTP response. As a result, we receive ascii integers as response to our invoke-webrequest:
![image](https://user-images.githubusercontent.com/47246332/213874168-ad180684-b626-4c11-85e3-f364339d732c.png)

In order to fix it I suggest we add text/html; charset=UTF-8 to the content-type header so that we receive the config in normal utf-8 text.

Reply:
The default text/html is incorrect, as the content returned by the intermediate server is _not_ HTML. The PowerShell script can be changed to interpret `application/x-wireguard-profile` and `application/x-openvpn-profile` correctly. Here's the snippet from my PowerShell that works fine with the proper Content-Type header:

```powershell
    if ($Response.StatusCode -eq 200) {
        # Using winget as system user is quite a hassle (https://github.com/microsoft/winget-cli/issues/548).
        # It can not find the winget path by itself so we need to resolve the path
        # (https://call4cloud.nl/2021/05/cloudy-with-a-chance-of-winget/#part3):

        # $ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
        # if ($ResolveWingetPath) {
        #     $WingetPath = $ResolveWingetPath[-1].Path
        # }
        # else {
        #     Throw "Winget not found"
        # }
        # cd $WingetPath

        # Install and deploy WireGuard tunnel if we received a wireguard-configuration
        switch ($Response.Headers['Content-Type']) {
            'application/x-wireguard-profile' {
                # .\winget.exe install --silent WireGuard.WireGuard --accept-package-agreements --accept-source-agreements | Out-Null

                # We create a new folder in the WireGuard directory.
                # We cant put it in WireGuard\Data directory as that folder is created only when we start the WireGuard application
                # (https://www.reddit.com/r/WireGuard/comments/x6f1gl/missing_data_directory_when_installing_wireguard/)
                New-Item -Path 'C:\Program Files\WireGuard' -Name 'eduVpnProvisioning' -ItemType 'directory' -ErrorAction Ignore

                # Limit access to the System user and administrators.
                icacls 'C:\Program Files\WireGuard\eduVpnProvisioning' /inheritance:r /grant:r 'SYSTEM:(OI)(CI)F' /grant:r 'Administrators:(OI)(CI)F'

                $Utf8NoBomEncoding = New-Object System.Text.UTF8Encoding $False
                [System.IO.File]::WriteAllLines('C:\Program Files\WireGuard\eduVpnProvisioning\wg0.conf', "$Response", $Utf8NoBomEncoding)
                & 'C:\Program Files\WireGuard\wireguard.exe' /installtunnelservice 'C:\Program Files\WireGuard\eduVpnProvisioning\wg0.conf'
            }
            'application/x-openvpn-profile' {
                # .\winget.exe install OpenVPNTechnologies.OpenVPN --silent --accept-package-agreements --accept-source-agreements --override "ADDLOCAL=OpenVPN.Service,OpenVPN,Drivers.TAPWindows6,Drivers" | Out-Null
                icacls "C:\Program Files\OpenVPN\config-auto" /inheritance:r /grant:r "SYSTEM:(OI)(CI)F" /grant:r "Administrators:(OI)(CI)F"
                $Utf8NoBomEncoding = New-Object System.Text.UTF8Encoding $False
                [System.IO.File]::WriteAllLines('C:\Program Files\OpenVPN\config-auto\openvpn.ovpn', "$Response", $Utf8NoBomEncoding)
                Restart-Service -Name OpenVPNService
            }
            default {
                Throw "Unexpected response: $($Response.Headers['Content-Type'])`r`n$Response"
            }
        }
```

Hadn't had time to finish reworking of the script generation (remember our talk on Signal the other day). Hence, this has not been added to this PR yet.

Nevertheless, I really suggest we send correct Content-Type header and fix the problem on the receiver side: namely PowerShell.

Reply:
Alright if that is the convention then lets do that 👍